src/mechanics/mesh/UnitMeshFem.cpp

```diff
@@ -60,6 +60,8 @@ MeshFem UnitMeshFem::CreateQuads(int numX, int numY)
 
 void UnitMeshFem::Transform(MeshFem* rMesh, std::function<Eigen::VectorXd(Eigen::VectorXd)> f)
 {
-    auto nodes = rMesh->NodesTotal();
+    // Build a group to avoid duplicates. Otherwise, the transformation is applied multiple times.
+    // This is, however, a bit of a bottleneck for big meshes. 
+    Groups::Group<NodeSimple> nodes = rMesh->NodesTotal();
```

Comment:
I don't fully understand. What is the difference between `mesh.Nodes` and `mesh.NodesTotal()`? What *duplicates* would show up *otherwise*?

Reply:
Definitely a point for the renaming issue. `NodesTotal()` returns all (unique - because of the Group) coordinate nodes. In contrast to `NodesTotal(dofType)` that returns all unique nodes for `dofType`.